EANvalidator/src/main/java/org/fundacionjala/CodingWar/EANValidator.java

```diff
@@ -0,0 +1,81 @@
+package org.fundacionjala.CodingWar;
+
+/**
+ * Created by ErickaViraca on 8/26/2016.
+ */
+public class EANValidator {
+
+    /**
+     * Validates the EAN string
+     *
+     * @param EAN, string of value
+     * @return --> true or false
+     */
+    public static boolean validate(String EAN) {
+
+        boolean res;
+        int checksum = 0;
+        int totalSum = additionResults(EAN);
+        final int divisor = 10;
+        final int checksumPosition = 12;
+
+        if (totalSum % divisor == 0) {
+            res = verifyChecksum(checksum, Integer.parseInt(String.valueOf(EAN.charAt(checksumPosition))));
+        } else {
+            checksum = calculateNewChecksum(totalSum);
+            res = verifyChecksum(checksum,  Integer.parseInt(String.valueOf(EAN.charAt(checksumPosition))));
+        }
+        return res;
```

Comment:
A lot of code here. and duplication too.
Try to apply the following lines of code.

```java
public static boolean validate(String EAN) {
    int totalSum = additionResults(EAN);
    final int divisor = 10;
    int checksum = totalSum % divisor == 0 ? 0 : divisor - (totalSum % divisor);
    return verifyChecksum(checksum,  Integer.parseInt(String.valueOf(EAN.charAt(EAN.length() - 1))));
}
```

and also you new to remove calculateNewChecksum() method as well.